src/tests/commitments.rs

```diff
@@ -0,0 +1,659 @@
+                0xbf, 0x8b, 0x88, 0x3a, 0x3e, 0x95, 0xb7, 0x98, 0x9c, 0xad, 0x19, 0x70, 0x63, 0x14,
+                0x1e, 0xbb, 0xfd, 0x00,
+            ],
+            ovk: [
+                0x57, 0x34, 0x67, 0xa7, 0xb3, 0x0e, 0xad, 0x6c, 0xcc, 0x50, 0x47, 0x44, 0xca, 0x9e,
+                0x1a, 0x28, 0x1a, 0x0d, 0x1a, 0x08, 0x73, 0x8b, 0x06, 0xa0, 0x68, 0x4f, 0xea, 0xcd,
+                0x1e, 0x9d, 0x12, 0x6d,
+            ],
+            ak: [
+                0x71, 0xc3, 0x52, 0x3e, 0xec, 0xa3, 0x53, 0x11, 0xfb, 0xd5, 0xd7, 0xe7, 0xd7, 0x0b,
+                0x70, 0x9d, 0x6c, 0x35, 0xa2, 0x4f, 0x26, 0x2b, 0x34, 0xbf, 0x64, 0x05, 0x9b, 0xf2,
+                0xc0, 0x2e, 0x0b, 0xa8,
+            ],
+            nk: [
+                0x62, 0x44, 0x00, 0x10, 0x3b, 0x65, 0x69, 0xb7, 0x35, 0x8f, 0xe8, 0x0f, 0x6f, 0x6c,
+                0xad, 0x43, 0x25, 0xde, 0xfd, 0xa9, 0xd9, 0x49, 0x9c, 0x2b, 0x8f, 0x88, 0x6a, 0x62,
+                0x69, 0xa2, 0xaa, 0x52,
+            ],
+            ivk: [
+                0xdb, 0x95, 0xea, 0x8b, 0xd9, 0xf9, 0x3d, 0x41, 0xb5, 0xab, 0x2b, 0xeb, 0xc9, 0x1a,
+                0x38, 0xed, 0xd5, 0x27, 0x08, 0x3e, 0x2a, 0x6e, 0xf9, 0xf3, 0xc2, 0x97, 0x02, 0xd5,
+                0xff, 0x89, 0xed, 0x00,
+            ],
+            default_d: [
+                0x23, 0x3c, 0x4a, 0xb8, 0x86, 0xa5, 0x5e, 0x3b, 0xa3, 0x74, 0xc0
+            ],
+            default_pk_d: [
+                0xb6, 0x8e, 0x9e, 0xe0, 0xc0, 0x67, 0x8d, 0x7b, 0x30, 0x36, 0x93, 0x1c, 0x83, 0x1a,
+                0x25, 0x25, 0x5f, 0x7e, 0xe4, 0x87, 0x38, 0x5a, 0x30, 0x31, 0x6e, 0x15, 0xf6, 0x48,
+                0x2b, 0x87, 0x4f, 0xda,
+            ],
+            note_v: 17811330145809239872,
+            note_r: [
+                0x6e, 0xbb, 0xed, 0x74, 0x36, 0x19, 0xa2, 0x56, 0xf9, 0xad, 0x2e, 0x85, 0x88, 0x0c,
+                0xfa, 0xa9, 0x09, 0x8a, 0x5f, 0xdb, 0x16, 0x29, 0x99, 0x0d, 0x9a, 0x7d, 0x3b, 0xb9,
+                0x3f, 0xc9, 0x00, 0x03,
+            ],
+            note_cm: [
+                0xd3, 0x76, 0xa7, 0xbe, 0xe8, 0xce, 0x67, 0xf4, 0xef, 0xde, 0x56, 0xaa, 0x77, 0xcf,
+                0x64, 0x41, 0x9b, 0x0e, 0x55, 0x0a, 0xbb, 0xcb, 0x8e, 0x2b, 0xcb, 0xda, 0x8b, 0x63,
+                0xe4, 0x1d, 0xeb, 0x37,
+            ],
+            note_pos: 2578461368,
+            note_nf: [
+                0x65, 0x36, 0x74, 0x87, 0x3b, 0x3c, 0x67, 0x0c, 0x58, 0x85, 0x84, 0x73, 0xe7, 0xfe,
+                0x72, 0x19, 0x72, 0xfb, 0x96, 0xe2, 0x15, 0xb8, 0x73, 0x77, 0xa1, 0x7c, 0xa3, 0x71,
+                0x0d, 0x93, 0xc9, 0xe9,
+            ],
+        },
+    ];
+
+    for tv in test_vectors {
+        // Compute commitment and compare with test vector
+        let mut result = [0u8; 32];
+        assert!(librustzcash_sapling_compute_cm(
+            &tv.default_d,
+            &tv.default_pk_d,
+            tv.note_v,
+            &tv.note_r,
+            &mut result
+        ));
+        assert_eq!(&result, &tv.note_cm);
```

Comment:
I would have just put this in `src/tests/key_components.rs`, rather than duplicating the test vectors. But we can clean this up later.